book/4-processing.md

```diff
@@ -13,3 +13,315 @@ kernelspec:
 ---
 
 # Parallel processing
+
++++
+
+In this chapter we'll look at how to efficiently process Zarr arrays.
+We'll do it with a focus on scalability.
+Although the examples here below operate on small arrays, the theory and techniques described will scale to processing large datasets and machines with more resources (e.g., number of processing cores).
+
+When we process an array, we are transforming it in some way and creating a new array - think of this as a mapping from one array to another.
+Because a Zarr array is split into chunks, we can split up this mapping into many smaller jobs.
+When writing a Zarr array whole chunks are compressed at written in one go, so it makes sense to break down the task into a mapping a mapping from the input array to each individual output chunk, and then create one job for each output chunk.
+This will allow us to compute each output chunk independently, which has two large benefits:
+
+- For each job we only need to load the data into memory needed to compute a single output chunk
+- We can parallelise the computation of output chunks, speeding up the whole process.
+
+To start with lets look at the simplest possible mapping from input chunks to output chunks - where the output array has the same shape and chunk size as the input array, and each output chunk depends only on the corresponding input chunk.
+
++++
+
+## Element-wise operations
```

Comment:
I am probably nitpicking, so feel free to ignore entirely. Is it worth introducing the term "element-wise" at all? Could it just be called "chunk-wise" and make things simpler? I don't think there's any mention of what is meant by "element" anywhere.